src/transformation/visitors/binary-expression/compound.ts

```diff
@@ -146,27 +140,39 @@ export function transformCompoundAssignmentExpression(
         const assignStatements = transformAssignment(context, lhs, tmpIdentifier);
 
         if (isSetterSkippingCompoundAssignmentOperator(operator)) {
-            return createImmediatelyInvokedFunctionExpression(
-                [tmpDeclaration, ...transformSetterSkippingCompoundAssignment(context, tmpIdentifier, operator, rhs)],
-                tmpIdentifier,
-                expression
-            );
+            const statements = [
+                tmpDeclaration,
+                ...transformSetterSkippingCompoundAssignment(context, tmpIdentifier, operator, rhs),
+            ];
+            return { statements, result: tmpIdentifier };
         }
 
-        return createImmediatelyInvokedFunctionExpression(
-            [tmpDeclaration, ...assignStatements],
-            tmpIdentifier,
-            expression
-        );
+        return { statements: [tmpDeclaration, ...assignStatements], result: tmpIdentifier };
     } else {
         // Simple expressions
         // ${left} = ${right}; return ${right}
         const operatorExpression = transformBinaryOperation(context, left, right, operator, expression);
-        const assignStatements = transformAssignment(context, lhs, operatorExpression);
-        return createImmediatelyInvokedFunctionExpression(assignStatements, left, expression);
+        const statements = transformAssignment(context, lhs, operatorExpression);
+        return { statements, result: left };
     }
 }
 
+export function transformCompoundAssignmentExpression(
+    context: TransformationContext,
+    expression: ts.Expression,
+    // TODO: Change type to ts.LeftHandSideExpression?
```

Comment:
Is this a question or a todo? Either way I think we should do a separate PR where we address all the TS depuration issues. IRRC we have a few places where we use old API.

Reply:
Should I remove this? It was just copied over from where the code was previously.

Reply:
Oh, in that case just keep it. Until we figure out what exactly was meant by this.

Reply:
Yeah keep it for now.